lib/rtoolsHCK.rb

```diff
@@ -345,61 +327,208 @@ def deploy_tools_hck(script_file)
     logger('debug', 'initialize/toolsHCK') { 'toolsHCK deployed on guest!' }
   end
 
+  # toolsHCK server TCP socket listener's port base
+  TOOLSHCK_TCP_SOCKET_PORT_BASE = 4000
+
+  # toolsHCK server TCP socket listener's port range
+  TOOLSHCK_TCP_SOCKET_PORT_RANGE = 1000
+
+  # toolsHCK server TCP socket listener's timeout
+  TOOLSHCK_TCP_SOCKET_TIMEOUT = 60
+
   def load_toolshck
-    load_toolshck_telnet
-    load_toolshck_shell
+    load_toolshck_server
+    load_toolshck_tcp_socket
   rescue StandardError
     @winrm_ps.close
     raise
   end
 
-  # Telnet's prompt match, (regex)
-  TELNET_PROMPT_MATCH = /^.:.*>/.freeze
+  def toolshck_server_job_script(tmp_rpath)
+    'Start-Job -ScriptBlock { powershell -ExecutionPolicy Bypass -File '\
+    "C:\\toolsHCK.ps1 -server -timeout #{TOOLSHCK_TCP_SOCKET_TIMEOUT} -port "\
```

Comment:
Let's have the path as constant. Also might be a good idea to move to configuration